### ISSUE TITLE
Add electric meter id number on options JSON object

### ISSUE THREAD

**Issue by peter279k** [peter279k/green-button-converter#15]
As title. And this number should present on `title` tag in Green Button Data response.

**Comment by dfcoffin** [comment]
NAESB considers the Electric Meter ID as a potential security issue as it can be used to obtain Personal Identifiable Information (PII).  However, the Green Button Alliance OpenADE Task Force determined that to be properly used, energy data needs to provide some PII, such as Meter ID, site address, etc.  Therefore, NAESB REQ.21 ESPI Version 3.3 (published January 30, 2020) contains a Retail Customer XSD file in which the Meter ID is provided (the element <serialNumber> is part of the Meter Resource).

**Comment by peter279k** [comment]
Hi @dfcoffin, thanks for your reply.

Do you have any latest sample that you mention about new approach to let Green Button Data contains meter ID with PII?

**Comment by peter279k** [comment]
At this moment, I let the meter ID contain in this `<title>GreenButton User XXXX Feed</title>` on Green Button Data XML.

If any security approach or standard are released, I will be happy to do improvements :).

**Comment by dfcoffin** [comment]
The latest schema (Version 3.3) can be retrieved from NAESB:

[espi.xsd](https://naesb.org//copyright/espi.xsd)
[retailcustomer.xsd](https://naesb.org//copyright/customer.xsd)

**Comment by dfcoffin** [comment]
If XXXX is the Meter ID or Account Number the <title> element violates the PII exclusion of the standard.

**Comment by peter279k** [comment]
Hi @dfcoffin, thanks for your reply. You're right.

Actually, it seems that the Green Button Data XML doesn't contain any PII information.

I need to let PII information be in Data XML and let specific customer can identify.

I also figure out approach is as follows:

- Separate two XML data. One is Green Button XML data with no PII information.
And another is customer with [NAESB customer schema](https://naesb.org//copyright/customer.xsd).

Is this a proper way to do this? Does the NAESB provide a validating customer data service?

**Comment by dfcoffin** [comment]
Yes, that is the correct manner to provide PII information, in two separate data streams. Both support the UsagePoint, which is then used to connect the PII data with the Energy Data. 

NAESB does NOT provide a validating customer data service, but the [Green Button Alliance](https://www.greenbuttonalliance.org/testing), Inc. provides a Certification program.

**Comment by peter279k** [comment]
Hi @dfcoffin, thanks fot your reply. I also find the retail customer data with [PGERetailCustomer.xml](https://github.com/VerdigrisTech/green-button-data/blob/master/spec/fixtures/PGERetailCustomer.xml) and [BulkSCERetailCustomer.xml](https://github.com/VerdigrisTech/green-button-data/blob/master/spec/fixtures/BulkSCERetailCustomer.xml).

I think these two XML samples are good enough for me to accomplish customer data generating.

In `BulkSCERetailCustomer.xml`, I can let meter id in following XML snippets:

```
<content>
      <cust:CustomerAccount>
        <cust:name>2-00-111-1111</cust:name>
      </cust:CustomerAccount>
    </content>
```

In `PGERetailCustomer.xml`, I can use following XML snippets:

```
 <ns0:CustomerAgreement xmlns:ns0="http://naesb.org/espi/customer">
        <ns0:name>this-is-meter-id</ns0:name>
        <ns0:signDate>1376550000</ns0:signDate>
      </ns0:CustomerAgreement>
```

What do you think about this idea? Thanks :).

And another question is: Do you have any good tutorial about how to let my data generator apply the 
Green Button Testing & Verification Program?

**Comment by dfcoffin** [comment]
The PGERetailCustomer.xml is based on the XSD released in July.  The most recent was released on January 30, 2020 and deprecated the <name> element because the PGERetailCustomer implementation was using <name> for numerous data elements withing multiple Retail Customer Resources.  The latest version of the XSD provides specific elements for example, <accountId> replaces <name> in the <CustomerAccount> structure:

```
<content>
    <cust:CustomerAccount>
        <accountId>2-00-111-1111</cust:accountId>
    </cust:CustomerAccount>
 </content>

```
If you compare the PGERetailCustomer XSD with the NAESB version (see link above) you will be able to identify the minor differences, but the new elements will make it easier for applications receiving the data to find what the data that want.

**Comment by peter279k** [comment]
Closed via PR #16.
